### PR TITLE
Automagic dashboards i18n

### PR DESCRIPTION
**Pull request by sbelak**
This adds a new lein task `generate-automagic-dashboards-pot` that walks all rules, extracts strings that should be translated as per `Rules` schema and dumps them into `/locales/metabase-automagic-dashboards.pot` 

~~Things I'm unsure about:~~
~~* is such a rudimentary .pot fine, or should it include metadata like content type and such?~~
~~* having line numbers or at least file numbers would be nice. Can we make do without for the initial version?~~
~~* Is it enough to just pipe the pot into metabase.pot in /bin/i18n/update-translation-template or are other steps needed?~~

### CODE REVIEW

**Comment by tlrobinson** [comment]
@sbelak This seems like a good start. I pushed a commit that integrates it with our existing tooling, so the workflow won't have to change.

A couple things:

1. it looks like no header is working fine. `msgcat` retains the ones from `metabase-frontend.pot` and `metabase-backend.pot`
2. yes line numbers would be nice, but even just filenames or a placeholder like `#: resources/automagic_dashboards/*.yaml` would probably be useful to translators so they know approximately where the string came from.
3. your template variable format is different than our others (which are just positional like `{0}`) but I don't think that's a problem as long as translators are careful not to change the template variable name.

It would be good to get someone to try this out end-to-end.

**Comment by sbelak** [comment]
3) is intentional so you can have a heuristic generate a SQL template and still inject bindings from the heuristic.

**Comment by sbelak** [comment]
Tried it, resulting .po looks reasonable (now also includes from where it is). I'm erring on the side of caution and disambiguating the same string in different heuristics (via msgctxt). Not sure if this is the right thing to do, or should we just yolo them all together.

**Review by tlrobinson — CHANGES_REQUESTED** [review]
I can't get this to work. Here's what I'm doing:

1. run `./bin/i18n/update-translations`, verify `locales/metabase.pot` and `locales/es.po` contain auto dashboard strings
2. localize an automatic dashboard string in `locales/es.po`
3. run `./bin/i18n/build-translation-resources`
4. start the server using `yarn dev`
5. set my browser's preferred language to Spanish, and also set the language in http://localhost:3000/admin/settings/general (just in case it uses `trs` instead of `tru`)
6. open Metabase in the browser, verify the frontend (except auto dashboards) is in Spanish
7. view an automatic dashboard, notice the string I translated above is still showing English rather than the translation

**Comment by mazameli** [comment]
I'll go ahead and be the nag on this. Is this the correct order of operations to close this out?

1. Get code review / approval from @tlrobinson on this?
2. Merge this into the `release-0.31.0` branch.
3. I'll then extract all the strings from the release branch, push that to POEditor, and then ping our translators.

Want to get this closed out ASAP so that we can get updated translations, otherwise it's going to hold up releasing 0.31.

**Comment by sbelak** [comment]
Looks about right. The issue is that it's a bit of a chicken and egg. We won't know if anything fell through the cracks until we get the translations completed.

File: src/metabase/automagic_dashboards/core.clj
Comment (senior) [inline src/metabase/automagic_dashboards/core.clj]: Looks like there is supposed to be two parameters  here but I only see one.

File: test/metabase/automagic_dashboards/core_test.clj
Comment (senior) [inline test/metabase/automagic_dashboards/core_test.clj]: Any reason not to
```
(map #(str (magic/humanize-datetime dt %)) [:minute :day :week...])
```

**Review by senior — CHANGES_REQUESTED** [review]
Looks good other than the one comment where it looks like two parameters should be there, but there is only one.

File: src/metabase/automagic_dashboards/core.clj
Comment (sbelak) [reply src/metabase/automagic_dashboards/core.clj]: The second one is provided by cond->> (perhaps not the clearest idiom, granted)

**Pull request merged**
